lambda_function.py

```diff
@@ -0,0 +1,79 @@
+import os
+import re
+
+from nypl_py_utils.functions.log_helper import create_log
+from nypl_py_utils.functions.config_helper import load_env_file
+from record_processor import RecordProcessor
+
+
+def lambda_handler(event, context):
+    logger = create_log("lambda_function")
+    load_env_file(os.environ["ENVIRONMENT"], "config/{}.yaml")
+
+    logger.info("Starting event processing...")
+
+    if event is None:
+        logger.error("Event is undefined.")
+        # TODO: raise exception here?
+    else:
+        # All records under one event will have the same schema
+        schema_name = _pull_schema_name(event)
+        os.environ["SCHEMA_NAME"] = schema_name
+        schema_url = (
+            os.environ["NYPL_DATA_API_BASE_URL"]
+            + os.environ["SCHEMA_PATH"]
+            + f"{schema_name}"
+        )
+        output_format = "json" if schema_name != "LocationHours" else "csv"
+
+        processor = RecordProcessor(schema_url)
+        successes, failures = 0, 0
+        processed_records = []
+
+        try:
+            # In case of key capitalization
+            lowercase_event = {k.lower(): v for k, v in event.items()}
+
+            for record in lowercase_event["records"]:
+                if "data" in record:
+                    result = processor.process_record(record, output_format)
+                    if "ProcessingFailed" in result["result"]:
+                        failures += 1
```

Comment:
Log an error here. It's probably good to try the rest of the records anyway but we should get notified when the processing fails on even a single record.